src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/Visitors/CSharpTagHelperAttributeValueVisitor.cs

```diff
@@ -0,0 +1,193 @@
+
+using System;
+using Microsoft.AspNet.Razor.Parser;
+using Microsoft.AspNet.Razor.Parser.SyntaxTree;
+
+namespace Microsoft.AspNet.Razor.Generator.Compiler.CSharp
+{
+    /// <summary>
+    /// <see cref="CodeVisitor{CSharpCodeWriter}"/> that writes code for a non-<see langword="string"/> tag helper
+    /// bound attribute value.
+    /// </summary>
+    /// <remarks>
+    /// Since attribute value is not written out as HTML, does not emit instrumentation. Further this
+    /// <see cref="CodeVisitor{CSharpCodeWriter}"/> writes identical code at design- and runtime.
+    /// </remarks>
+    public class CSharpTagHelperAttributeValueVisitor : CodeVisitor<CSharpCodeWriter>
+    {
+        private const string DefaultPseudoMethod = "default";
+
+        private string _attributeTypeName;
+        private bool firstChild;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CSharpTagHelperAttributeValueVisitor"/> class.
+        /// </summary>
+        /// <param name="writer">The <see cref="CSharpCodeWriter"/> used to write code.</param>
+        /// <param name="context">
+        /// A <see cref="CodeBuilderContext"/> instance that contains information about the current code generation
+        /// process.
+        /// </param>
+        /// <param name="attributeTypeName">
+        /// Full name of the property <see cref="Type"/> for which this
+        /// <see cref="CSharpTagHelperAttributeValueVisitor"/> is writing the value.
+        /// </param>
+        public CSharpTagHelperAttributeValueVisitor(
+            CSharpCodeWriter writer,
+            CodeBuilderContext context,
+            string attributeTypeName)
+            : base(writer, context)
+        {
+            _attributeTypeName = attributeTypeName;
+        }
+
+        /// <summary>
+        /// Writes code for the given <paramref name="chunk"/>.
+        /// </summary>
+        /// <param name="chunk">The <see cref="ChunkBlock"/> to render.</param>
+        /// <remarks>
+        /// <para>
+        /// Write a line pragma and the code for all children. Writes <c>default(property type)</c> if
+        /// <paramref name="chunk"/> has no children. Pads the first code block to align with source.
+        /// </para>
+        /// <para>Also tracks code mappings for all children.</para>
+        /// </remarks>
+        protected override void Visit(ChunkBlock chunk)
+        {
+            var currentIndent = Writer.CurrentIndent;
+            try
+            {
+                // Entire attribute value is an expression. Write line number pragma for it.
+                using (var mappingWriter = new CSharpLineMappingWriter(Writer, chunk.Start, Context.SourceFile))
+                {
+                    firstChild = true;
+                    Accept(chunk.Children);
+
+                    if (firstChild)
+                    {
+                        // Attribute value was empty.
+                        Writer
```

Comment:
I'm not sure this is the write solution for an empty attribute because we dont want it to compile if they haven't provided anything. I'd say just do nothing, that way it doesn't compile and you'll get a reasonable C# error as to why it's not compiling. Also, doing `default(foo)` doesn't work when you have attribute types such as `Foo<T>` because the attribute type name comes through as `Foo'T`.